### DISCUSSION TITLE
how to upload language metadat for arabic language in appstore <ar> tag not supported in fatslane

### DISCUSSION THREAD

**Discussion by Computeeeeeeeeee** [fastlane/fastlane#28049] [General]
how to upload language metadat for arabic language in appstore <ar> tag not supported in fatslane

**Comment by jackdiz543** [comment]
Hello,

Steps to Upload Arabic Metadata in Fastlane
Verify the Language Code for Arabic: Apple uses language codes in *.json files or metadata. For Arabic, the code is:

Set Up Metadata Folder Structure: Fastlane organizes metadata in a specific structure under the fastlane/metadata directory. Create a folder for Arabic:     [SnapTik com](https://wwwsnaptik.com)

bash
Edit
fastlane/metadata/ar/
Add Metadata Files: Inside the ar folder, add the required metadata files. Examples include:

description.txt (App description in Arabic)
keywords.txt (Comma-separated Arabic keywords)
release_notes.txt (Release notes in Arabic)
marketing_url.txt, support_url.txt, and privacy_url.txt (URLs, if needed)
Screenshots (in the correct dimensions for the App Store)
Update Deliverfile or Fastlane Configuration: Make sure Fastfile or Deliverfile is configured to include the Arabic metadata. Example:

ruby
Edit
deliver(
  skip_binary_upload: true,
  skip_screenshots: false,
  skip_metadata: false
)
Verify Metadata Encoding: Ensure that all text files (.txt) are saved in UTF-8 encoding to properly support Arabic characters.

Run Fastlane Deliver Command: Execute the following command to upload metadata:

bash
fastlane deliver --verbose
This will upload the metadata and screenshots for all languages, including Arabic.

Debugging Unsupported Tag Errors: If you still encounter an error like "tag not supported", try the following:

Double-check the folder structure and ensure the ar folder exists under metadata.
Make sure you have the correct Arabic language code (ar).
Check your App Store Connect account to verify that Arabic is added as a localization language.
Manually Add Arabic in App Store Connect: If the language is not already available in App Store Connect, you must manually add Arabic as a supported language:

Go to App Store Connect > Your App > App Information > Localizations.
Add Arabic (ar) as a new language.
Rerun the Fastlane command.

Best Regards
 jack543
